### ISSUE TITLE
Feature request: configure default branch

### ISSUE THREAD

**Issue by tomsabin** [prose/prose#876]
By default, the selected branch upon loading Prose is set to GitHub's default branch. It would be useful to be able to set the default Prose branch independently to GitHub's default.

The situation I have is that the default branch on GitHub is `staging`, however content written on Prose should be deployed automatically to `production` - currently the user would have to select this branch in the side panel. However, if the user navigates away from the "new file" view after selecting a new branch, this selection is lost (see https://github.com/prose/prose/issues/640).

Ideally, I'd like to be able to specify something like this in the `_prose.yml` config file:

```YML
prose:
  default_branch: 'staging'
```

**Comment by yguezeldal** [comment]
+1

**Comment by catherine-jones** [comment]
+1

**Comment by catherine-jones** [comment]
@dereklieu happy to work on this

**Comment by klepas** [comment]
+1

**Comment by dereklieu** [comment]
@catherine-jones Awesome 👍 . One thing, by the time Prose [gets around to reading `_config.yml`](https://github.com/prose/prose/blob/master/app/collections/files.js#L77), it will already have chosen a [default branch](https://github.com/prose/prose/blob/master/app/views/repo.js#L27). I don't think we can get around loading the default branch according to Github before then loading the default branch set in `_config.yml`, if it exists.

The logic that loads that second branch would also have to be smart enough to correct the default branch, while not interfering with the ability to manually change branches.

Let me know if you want to talk this out more!

**Comment by catherine-jones** [comment]
@dereklieu yes would be great to chat more about this let mw know how, when would be a good time.

**Comment by huphtur** [comment]
Would it be possible to ignore certain branches? eg:
```
prose:
  ignore_branch: 
    - master
    - staging
```
or only include certain branches?
```
prose:
  branch: 
    - development
```

**Comment by huphtur** [comment]
hey @huphtur, you could simply add a specific `_prose.yml` file to branches that need to be ignored ;-P

**Comment by dominikwilkowski** [comment]
This is currently the only blocker for us to use prose. We can't have content editors commit to `master` directly. ;)
